ansible/roles/test/tasks/fdb.yml

```diff
@@ -73,3 +73,85 @@
   always:
     - name: clear FDB table
       command: sonic-clear fdb all
+
+- block:
+  - name: clear FDB table
+    command: sonic-clear fdb all
+
+  - name: Get ports in portchannels
+    set_fact:
+      portchannel_members_all: []
+  - set_fact:
+      portchannel_members_all: "{{ portchannel_members_all + item['members'] }}"
+    with_items: "{{ minigraph_portchannels.values() }}"
+
+  - name: Get DUT ports info
+    set_fact:
+      vlan_ports: "{%for port in minigraph_vlans['Vlan1000']['members'] %}{{minigraph_port_indices[port]}}{%if not loop.last%} {%endif%}{%endfor%}"
+      lag_ports: "{%for port in portchannel_members_all %}{{minigraph_port_indices[port]}}{%if not loop.last%} {%endif%}{%endfor%}"
+
+  - name: Initialize variables
+    set_fact:
+      dummy_mac_prefix: "02:11:22:33"
+
+  - name: "Set ptf facts"
+    set_fact:
+      ptf_test_name: FDB test
+      ptf_test_dir: ptftests
+      ptf_test_path: fdb_test.FdbConfigReloadTest
+      ptf_platform: remote
+      ptf_platform_dir: ptftests
+      ptf_test_params:
+        - testbed_type='{{ testbed_type }}'
+        - vlan_ports='{{ vlan_ports }}'
+        - lag_ports='{{ lag_ports }}'
+        - router_mac='{{ ansible_interface_facts['Ethernet0']['macaddress'] }}'
+        - dummy_mac_prefix='{{ dummy_mac_prefix }}'
+      ptf_extra_options: "--relax --debug info --log-file /tmp/fdb_test.FdbTest.{{lookup('pipe','date +%Y-%m-%d-%H:%M:%S')}}.log "
+
+  - debug: msg="ptf --test-dir {{ ptf_test_dir }} {{ ptf_test_path }} {% if ptf_qlen is defined %} --qlen={{ ptf_qlen }} {% endif %} {% if ptf_platform_dir is defined %} --platform-dir {{ ptf_platform_dir }} {% endif %} --platform {{ ptf_platform }} {% if ptf_test_params is defined %} -t \"{{ ptf_test_params | default([]) | join(';') }}\" {% endif %} {{ ptf_extra_options | default(\"\")}} 2>&1"
+
+  - name: Run PTF script to inject packets with different src MAC
+    shell: ptf --test-dir {{ ptf_test_dir }} {{ ptf_test_path }} {% if ptf_qlen is defined %} --qlen={{ ptf_qlen }} {% endif %} {% if ptf_platform_dir is defined %} --platform-dir {{ ptf_platform_dir }} {% endif %} --platform {{ ptf_platform }} {% if ptf_test_params is defined %} -t "{{ ptf_test_params | default([]) | join(';') }}" {% endif %} {{ ptf_extra_options | default("")}} 2>&1
+    args:
+      chdir: /root
+    delegate_to: "{{ ptf_host }}"
+    async: 300
+    poll: 0
+
+  - name: Reload config
+    command: config reload -y
+    become: yes
+
+  - name: Wait until the PTF script completed execution
+    shell: "ps -ef | grep /usr/bin/ptf | grep -v grep"
```

Comment:
>ps -ef | grep /usr/bin/ptf | grep -v grep [](start = 12, length = 41)

Could you try `wait pid` or `tail --pid=$pid -f /dev/null` ?

Reply:
OK, updated and comitted.